File: ta1.js
import { irradix, derradix, VALS } from './index.js'

let val = 10n;
for( let i = 0; i < 200; i++ ) {
  const basePhi = irradix(val, VALS.BigPHI);
  const regular = derradix(basePhi, VALS.BigPHI);
  if ( regular != val ) {
    console.log(`Divergence at ${i+2}-digit random value.`);
    const vs = val.toString();
    const rs = regular.toString();
    const track = [];
    for(let j = 0; j < Math.min(vs.length, rs.length); j++) {
      if ( vs[j] != rs[j] ) {
        console.log(`Values differ at the ${j+1}th place:`);
        track.push('^');
      } else {
        track.push(' ');
      }
    }
    console.log(`Original:  ${vs}`);
    console.log(`Recovered: ${rs}`);
    console.log(`           ${track.join('')}`);
  } else {
    console.log(`${i+2}-digit values match: ${regular}`);
  }

  const digit =  BigInt(Math.round(Math.random()*10));
  val *= 10n;
  val += digit;
}

File: ta1.py
#!/usr/bin/env python

from mpmath import mp
import random
from irradix import irradix, derradix, set_precision

set_precision(1000)
val = mp.mpf(10)

for i in range(1000):
    basePhi = irradix(val)
    regular = derradix(basePhi)
    if regular != int(val):
        print(f"Divergence at {i+2}-digit random value.")
        vs = str(int(val))
        rs = str(int(regular))
        track = []
        diff_places = []
        for j in range(min(len(vs), len(rs))):
            if vs[j] != rs[j]:
                diff_places.append(j+1)
                track.append('^')
            else:
                track.append(' ')
        print(f"Original:  {vs}")
        print(f"Recovered: {rs}")
        print(f"           {''.join(track)}")
        print(f"Values differ at places: {diff_places}")
        input("Press Enter to continue...")  # Pause for user input
    else:
        print(f"{i+2}-digit values match: {regular}")

    digit = random.randint(0, 9)
    val *= 10
    val += digit

